routers/web/repo/release.go

```diff
@@ -198,6 +199,12 @@ func Releases(ctx *context.Context) {
 
 // TagsList render tags list page
 func TagsList(ctx *context.Context) {
+	// The tag list page will be enabled only when user cannot visit releases page.
+	if ctx.Repo.CanRead(unit.TypeReleases) {
+		ctx.Redirect(fmt.Sprintf("%s/releases", ctx.Repo.RepoLink))
+		return
+	}
```

Comment:
It is completely wrong.

I had enough for reviewing your buggy code, do you really understand the problem when writing code? Maybe you should ask others to review first.

Reply:
51f4ad8